js/node-manager/src/client/screens/Settings.tsx

```diff
@@ -1,14 +1,31 @@
 import { useAppState } from '../app-state'
 import { Button } from '../components/basics'
-import { Input, Layout } from '../components/Layout'
+import { Layout } from '../components/Layout'
 import React, { useEffect, useMemo, useState } from 'react'
 import AceEditor from 'react-ace'
 import clsx from 'clsx'
 import { NodeType, ReachableNodeUi } from '../../common/types'
 import { get, set, parse } from 'json-pointer'
-import deepEqual from 'deep-equal'
 import { validateAgainstSchema } from '../components/SettingsEditor'
 
+// Handles case where `set` cannot set root path
+// Because require('json-pointer').set throws when `path` is empty
+const setOrReplace = (s: object, path: string, val: unknown): object => {
+  // parse:
+  //  "/a/b" => ['a', 'b']
+  //  "/" => []
+  //  "" => []
+  const pointer = parse(path.trim()).filter((x) => !!x)
```

Comment:
The `!!x` smells wrong to me: what if a path component is `'false'`?

Reply:
Per NodeJS 16

`!!false === false`
`!!'false' === true`

The issue where `'false'` string is regarded as falsy has been gone for I think years or decades.
(Even I am almost convinced that it is an instance of Mandela Effect because I can't find it anywhere online, but it might be there in PHP instead because a long time ago I am sure I had the same problem)

Do we use `'false'` string anywhere btw?

Reply:
Ah, I didn't realize that this indirection is confusing to some people.
I thought this implication was universal because '' is the only falsy string there is in JS.
Will change to that

Reply:
well, `null` or `undefined` are also falsy, IIRC

Reply:
I mean, falsy string since `parse` returns string[].
Fixed it